src/Compiler/CompilerUtils.cpp

```diff
@@ -490,6 +490,16 @@ static int genSharedLib(std::string sharedLibNameWithExt,
     std::vector<std::string> opts, std::vector<std::string> objs,
     std::vector<std::string> libs, std::vector<std::string> libDirs) {
 
+  // Add user specified libs and their path
+  // Multiple lib or directory can be specified with multiple options.
+  // For example, --extr-libs="extra1", --extra-libs="extra2"
+
+  for (std::string lib : extraLibs)
+    libs.push_back(lib);
+
+  for (std::string dir : extraLibDirs)
+    libDirs.push_back(dir);
+
```

Comment:
Instead of changing `genSharedLib`, try to use the existing `addCompilerConfig`/`getCompilerConfig` routines like this: https://github.com/onnx/onnx-mlir/blob/76a3f070282806cbd4b6468c2b8f9ebafece554b/src/Compiler/CompilerUtils.cpp#L761

The `CCM_SHARED_LIB_DEPS` key already covers your `-l` option. For `-L`, you can define another key, for example,
```
#define CCM_SHARED_LIB_DEPS "sharedLibDirs"
```
to collect all the directories specified.

Reply:
@gongsu832 This seems to be a better solution. I made the change and please review it.

Reply:
LGTM. Thanks.